src/rust/include/rust/zip339.h

```diff
@@ -0,0 +1,59 @@
+#ifndef ZCASH_RUST_INCLUDE_RUST_ZIP339_H
+#define ZCASH_RUST_INCLUDE_RUST_ZIP339_H
+
+#include "rust/types.h"
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifndef __cplusplus
+  #include <assert.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    // These must match src/rust/src/zip339_ffi.rs.
+    // (They happen to also match the integer values correspond in the bip0039-rs crate with the
+    // "all-languages" feature enabled, but that's not required.)
+    enum Language
+#ifdef __cplusplus
+    : uint32_t
+#endif
+    {
+        English = 0,
+        SimplifiedChinese = 1,
+        TraditionalChinese = 2,
+        Czech = 3,
+        French = 4,
+        Italian = 5,
+        Japanese = 6,
+        Korean = 7,
+        Portuguese = 8,
+        Spanish = 9,
+        SIZE_HACK = 0xFFFFFFFF // needed when compiling as C
+    };
+
+    static_assert(sizeof(Language) == 4);
+
+    /// Creates a phrase with the given entropy, in the given `Language`. The phrase is represented as
+    /// a pointer to a null-terminated C string that must not be written to, and must be freed by
+    /// `zip339_free_phrase`.
+    const char * zip339_entropy_to_phrase(Language language, const uint8_t *entropy, size_t entropy_len);
```

Comment:
This needs to return a mutable type, since the Rust side is transferring ownership, and the return type here needs to match the parameter type of `zip339_free_phrase`:
```suggestion
    char * zip339_entropy_to_phrase(Language language, const uint8_t *entropy, size_t entropy_len);
```

Reply:
Hmm. The intent is that the caller not mutate the string until it is freed. So it's correct on the C side to cast away const when calling `zip339_free_phrase` (as you typically have to do when calling `free` in C). I could have made `zip339_free_phrase` take a `const char *` but then I'd still have had to cast away const on the Rust side in an unsafe block, and I'm less sure of the semantics of that.